x-pack/plugins/maps/public/classes/layers/file_upload_wizard/config.tsx

```diff
@@ -9,12 +9,24 @@ import { i18n } from '@kbn/i18n';
 import React from 'react';
 import { LayerWizard, RenderWizardArguments } from '../../layers/layer_wizard_registry';
 import { ClientFileCreateSourceEditor, INDEX_SETUP_STEP_ID, INDEXING_STEP_ID } from './wizard';
+import { getFileUpload } from '../../../kibana_services';
 
 export const uploadLayerWizardConfig: LayerWizard = {
   categories: [],
   description: i18n.translate('xpack.maps.fileUploadWizard.description', {
     defaultMessage: 'Index GeoJSON data in Elasticsearch',
   }),
+  disabledReason: i18n.translate('xpack.maps.fileUploadWizard.disabledDesc', {
+    defaultMessage:
+      'You do not have the required file upload privileges. You are missing the Kibana privilege "Index Pattern Management".',
```

Comment:
Why is the `disabledReason` hard-coded to "Index Pattern Management"? Shouldn't we return the reason from the endpoint, save it here, and then provide it in this message?

Reply:
I am modeling the endpoint after the ml endpoint which just return a boolean. Also, in this instance, there is no way to get the results from the endpoint since disabledReason is not async.

Reply:
Should we be more generic in our messaging then? We don't really know which privilege the user doesn't have here and this message might be misleading. For instance they might not have `create index` privs. The two sentences are also a little redundant. Maybe replace the whole thing with something simple like `You are missing one or more of the required File upload privileges.` Eventually we can just link to documentation for further details.

Reply:
This call to hasImportPermissions occurs before the user has entered an index name so the only permission that is getting checked is to see if the user has permission to create an index pattern. I can clean up and put into a single sentence.